staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -653,6 +653,8 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 		return newErrWatcher(err), nil
 	}
 
+	c.setBookmarkNewFuncIfRequested(cacheInterval, opts)
```

Comment:
Let's call this function `setInitialEventsEndBookmarkIfRequested`

Reply:
SGTM